gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java

```diff
@@ -261,4 +284,31 @@ public void showSettings() {
   public void onGeoJsonClick(MapGeoJson mapGeoJson) {
     showBottomSheet(mapGeoJson.getFeature());
   }
+
+  public Observable<ImmutableList<Feature>> getCandidateFeatures() {
+    return candidateFeatures;
+  }
+
+  public void onFeatureClick(ImmutableList<MapFeature> mapFeatures) {
+    showFeatureSelector(mapFeatures);
+  }
+
+  private void showFeatureSelector(ImmutableList<MapFeature> mapFeatures) {
+    ImmutableList<Feature> features =
+        stream(mapFeatures)
+            .map(
+                feature -> {
+                  if (feature instanceof MapPin) {
+                    Feature feat = ((MapPin) feature).getFeature();
+                    return feat;
```

Comment:
We don't need the intermediate variable `feat` here and below, you can just `return ((cast)) feature)...`